src/GovUk.Education.ExploreEducationStatistics.Common/Extensions/HttpContextExtensions.cs

```diff
@@ -0,0 +1,29 @@
+#nullable enable
+using System.Diagnostics.CodeAnalysis;
+using Microsoft.AspNetCore.Http;
+
+namespace GovUk.Education.ExploreEducationStatistics.Common.Extensions;
+
+public static class HttpContextExtensions
+{
+    public static bool TryGetRequestHeader(
```

Comment:
I added this extension primarily to do a not-null check on `httpContext`  because it made things look a bit cleaner than adding a check to make sure `httpContextAccessor.HttpContext` is not null prior to calling `httpContextAccessor.HttpContext.Request.TryGetHeader` or assuming it's not null and using `httpContextAccessor.HttpContext!.Request.TryGetHeader`.

I used a not-null check before I added this extension but that code got repeated in both auth handlers.